runtime/vm/callin.cpp

```diff
@@ -362,6 +362,9 @@ buildCallInStackFrame(J9VMThread *currentThread, J9VMEntryLocalStorage *newELS,
 		}
 	}
 #endif /* J9VM_PORT_ZOS_CEEHDLRSUPPORT */
+#if JAVA_SPEC_VERSION >= 19
```

Comment:
I don't think you can do this here - every thread will always have at least a count of 1 if it's in the VM. Either move it to the resolve callers or to the specific callin helpers that are used (I greatly prefer the former).

Reply:
The way the code is now, the initial call-in when mounting a continuation for the first time would run with the counter set to 1, so it could never be unmounted.

You actually can do the management here as long as you do not inc/dec the counter on the first call in (vmThread->entryLoclStorage is NULL) and last return (will set mThread->entryLoclStorage  to NULL). This would be future-proof for the inevitable new call-in resolve cases that will come up.

Reply:
Updated.